### PR TITLE
Legends builder

### PR DESCRIPTION
**Pull request by nobuti**
This PR fine tunes the legends in the Builder:

- Fixes the form filling when rendering
- Fixes some issues about the integration with cartodb.js 
- Applied last changes in the vizjson in order to a correctly legend collection initialisation
- Avoid suggest legends if the user selected none previously

CR @xavijam

### CODE REVIEW

File: lib/assets/javascripts/cartodb3/data/legends/legend-bubble-definition-model.js [hunk 1]
```diff
@@ -1,23 +1,42 @@
 var _ = require('underscore');
 var LegendBaseDefModel = require('./legend-base-definition-model');
+var LegendColorHelper = require('../../editor/layers/layer-content-views/legend/form/legend-color-helper');
 
 module.exports = LegendBaseDefModel.extend({
   defaults: _.extend({}, LegendBaseDefModel.prototype.defaults,
     {
       type: 'bubble',
-      fillColor: '#fabada'
+      fillColor: null
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/data/legends/legend-bubble-definition-model.js]: No default color then!?

File: lib/assets/javascripts/cartodb3/deep-insights-integrations.js [hunk 2]
```diff
@@ -315,6 +312,8 @@ F.prototype._onLegendDefinitionAdded = function (m) {
   var legend;
   if (layer && layer.legends) {
     legend = layer.legends[type];
+    legend && legend.reset();
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/deep-insights-integrations.js]: I'd try with:
```js
if (legend) {
  …
}
```

File: lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/legend/form/legend-bubble-definition-form-model.js [hunk 3]
```diff
@@ -6,19 +6,25 @@ module.exports = LegendBaseDefModel.extend({
...
 
   initialize: function (attrs, opts) {
     this._titlePlaceholder = _t('editor.legend.legend-form.by-size');
     LegendBaseDefModel.prototype.initialize.call(this, attrs, opts);
 
+    // title
+    this.set(_.extend(
+      {},
+      _.pick(opts.legendDefinitionModel.attributes, 'title')
+    ), {silent: true});
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/legend/form/legend-bubble-definition-form-model.js]: It is weird to `set` the model in the initialize, couldn't we do it in the model creation?

File: lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/legend/form/legend-category-definition-form-model.js [hunk 4]
```diff
@@ -10,6 +10,13 @@ module.exports = LegendBaseDefModel.extend({
 
   initialize: function (attrs, opts) {
     this._titlePlaceholder = _t('editor.legend.legend-form.by-color');
+
+    // title
+    this.set(_.extend(
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/legend/form/legend-category-definition-form-model.js]: Same here ^^^

File: lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/legend/form/legend-choropleth-definition-form-model.js [hunk 5]
```diff
@@ -12,6 +12,13 @@ module.exports = LegendBaseDefModel.extend({
 
   initialize: function (attrs, opts) {
     this._titlePlaceholder = _t('editor.legend.legend-form.by-color');
+
+    // title, prefix, suffix
+    this.set(_.extend(
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/legend/form/legend-choropleth-definition-form-model.js]: Also here!

File: lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/legend/form/legend-custom-definition-form-model.js [hunk 6]
```diff
@@ -16,12 +16,30 @@ module.exports = LegendBaseDefModel.extend({
   initialize: function (attrs, opts) {
     this._titlePlaceholder = _t('editor.legend.legend-form.by-color');
     LegendBaseDefModel.prototype.initialize.call(this, attrs, opts);
-    this._inheritItems();
+
+    // title
+    this.set(_.extend(
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/legend/form/legend-custom-definition-form-model.js]: Dixit

Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/legend/form/legend-custom-definition-form-model.js]: Brexit

Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/legend/form/legend-custom-definition-form-model.js]: Spaixit?

File: lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/legend/legend-base-type-view.js [hunk 7]
```diff
@@ -37,9 +38,9 @@ module.exports = CoreView.extend({
     this._legendDefinitionModel = this._getUserLegend(this.legendTypes);
 
     var content = HTML_TEMPLATE({
-      preSnippet: this._legendDefinitionModel.get('pre_html') || '<!-- put your html above this line -->',
+      preSnippet: this._legendDefinitionModel.get('preHTMLSnippet') || '<!-- put your html above this line -->',
       placeholder: PLACEHOLDER,
-      postSnippet: this._legendDefinitionModel.get('post_html') || '<!-- put your html below this line -->'
+      postSnippet: this._legendDefinitionModel.get('postHTMLSnippet') || '<!-- put your html below this line -->'
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/legend/legend-base-type-view.js]: Shouldn't we translate those strings?

File: lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/legend/legend-properties-form.js [hunk 8]
```diff
@@ -23,7 +23,7 @@ module.exports = CoreView.extend({
       model: this._formModel
     });
 
-    this._formView.bind('change', this._updateChanges, this);
+    this._formView.on('change', this._updateChanges, this);
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/legend/legend-properties-form.js]: Is `add_related_model` needed here?

File: lib/assets/javascripts/cartodb3/data/legends/legend-bubble-definition-model.js (hunk 1 above)
Comment (nobuti) [inline lib/assets/javascripts/cartodb3/data/legends/legend-bubble-definition-model.js]: We are taking the color from styleModel when parsing the attributes.

File: lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/legend/form/legend-bubble-definition-form-model.js (hunk 3 above)
Comment (nobuti) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/legend/form/legend-bubble-definition-form-model.js]: Más razón que un santo

File: lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/legend/legend-properties-form.js (hunk 8 above)
Comment (nobuti) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/legend/legend-properties-form.js]: Good catch

**Pull request merged**